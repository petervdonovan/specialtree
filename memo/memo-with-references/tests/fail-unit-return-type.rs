use memo_with_references::memo;

#[memo('a)]
fn test_unit_return() -> () {
    println!("This has side effects and shouldn't be memoized");
}

fn main() {}
