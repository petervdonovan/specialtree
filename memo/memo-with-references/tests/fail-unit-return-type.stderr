error: memoized functions cannot have unit return type - there is no value to cache
 --> tests/fail-unit-return-type.rs:4:26
  |
4 | fn test_unit_return() -> () {
  |                          ^^
